dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild

```diff
@@ -50,6 +53,117 @@ python_prepare_all() {
 	if ! use opengl; then
 		rm -r pyqtgraph/opengl || die
 	fi
+
+	# pyqtgraph will automatically use any QT bindings it finds,
+	# patch the supported Qt versions to only include the ones we have enabled
+	local upstream_supported_qt=(
+		PySide
+		PySide2
+		PySide6
+		PyQt4
+		PyQt5
+		PyQt6
+	)
+	local gentoo_qt=()
+	local is_supported_impl use_qt qt
+	# It does comparisons using these variables; set deselected or unsupported to None to avoid more patching
+	for qt in "${upstream_supported_qt[@]}"; do
+		is_supported_impl=0
+		case ${qt} in
+			PySide6) is_supported_impl=1; use_qt=${qt} ;;
+			PyQt5) is_supported_impl=1; use_qt=${qt:2} ;;
+			PyQt6) is_supported_impl=1; use_qt=${qt:2} ;;
+		esac
+		if [[ "${is_supported_impl}" -eq 0 ]]; then
+			gentoo_qt+=( "${qt^^} = None" )
+			continue
+		fi
+		if use ${use_qt,,}; then
+			gentoo_qt+=( "${qt^^} = '${qt}'" )
+		else
+			gentoo_qt+=( "${qt^^} = None" )
+		fi
+	done
+
+	awk -v qt_string="$(printf "%s\n" "${gentoo_qt[@]}")" -i inplace '
+	BEGIN {
+		i = 0
+		split(qt_string, qt_array, "\n")
+		j = 1
+	}
+	/PYSIDE = '"'"'PySide'"'"'/ {
+		i = 6 # length of upstream_supported_qt
+		for (k = 1; k <= length(qt_array); k++) {
+		print qt_array[k]
+		j++
+		}
+	}
+	i > 0 {
+		i--
+		next
+	}
+	{ print }
+	' pyqtgraph/Qt/__init__.py || die "Failed to patch supported Qt versions"
+
+	# We also need to remove them from load order.
+	local liborder=()
+	local qt
+	# The order is important (we want to load the newest at runtime)
+	# so newest first here; check the file for the original order
+	for qt in qt6 pyside6 qt5; do
+		if use ${qt}; then
+			if [[ "${qt}" == qt* ]]; then
+				liborder+=( "PY${qt^^}" )
+			else
+				liborder+=( "${qt^^}" )
+			fi
+		fi
+	done
+
+	awk -v libOrder="$(printf "%s, " "${liborder[@]}")" -i inplace '
+	BEGIN {
+		libOrder = "[" substr(libOrder, 1, length(libOrder) - 2) "]"
+	}
+	/libOrder = \[PYQT6, PYSIDE6, PYQT5, PYSIDE2\]/ {
+		sub(/\[PYQT6, PYSIDE6, PYQT5, PYSIDE2\]/, libOrder)
+	}
+	{ print }
+	' pyqtgraph/Qt/__init__.py || die "Failed to patch qt version order"
```

Comment:
This looks fine, but it is rather complex, can't we do the same here as in QtPy.ebuild, i.e. sed the import with `raise ImportError` if the flag is disabled?